homeassistant/auth/providers/ldap.py

```diff
@@ -0,0 +1,252 @@
+            [CONF_ENCRYPTION_LDAPS, CONF_ENCRYPTION_NONE, CONF_ENCRYPTION_STARTTLS]
+        ),
+        vol.Required(CONF_PORT, default=DEFAULT_CONF_PORT): int,
+        vol.Required(CONF_SERVER): str,
+        vol.Required(CONF_TIMEOUT, default=DEFAULT_CONF_TIMEOUT): int,
+        vol.Required(CONF_USERNAME_ATTR, default=DEFAULT_CONF_USERNAME_ATTR): str,
+    },
+    extra=vol.PREVENT_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InvalidAuthError(HomeAssistantError):
+    """Raised when submitting invalid authentication."""
+
+
+class LdapError(HomeAssistantError):
+    """Raised when an LDAP error has been encountered."""
+
+
+@AUTH_PROVIDERS.register("ldap")
+class LdapAuthProvider(AuthProvider):
+    """LDAP auth provider."""
+
+    DEFAULT_TITLE = "LDAP Authentication"
+
+    async def async_login_flow(self, context: Optional[Dict]) -> LoginFlow:
+        """Return a flow to login."""
+        return LdapLoginFlow(self)
+
+    @callback
+    def async_validate_login(self, username: str, password: str) -> None:
+        """Validate a username and password."""
+        try:
+            tls = ldap3.Tls()
+            tls.validate = (
+                ssl.CERT_REQUIRED
+                if self.config[CONF_CERT_VALIDATION]
+                else ssl.CERT_NONE
+            )
+            tls.ca_certs_file = self.config[CONF_CA_CERTS_FILE]
+            encryption = self.config[CONF_ENCRYPTION]
+            # Server setup
+            server = ldap3.Server(
+                self.config[CONF_SERVER],
+                port=self.config[CONF_PORT],
+                use_ssl=encryption == CONF_ENCRYPTION_LDAPS,
+                tls=tls,
+                connect_timeout=self.config[CONF_TIMEOUT],
+                get_info=ldap3.ALL,
+            )
+
+            base_dn = self.config[CONF_BASE_DN]
+            bind_type = self.config[CONF_BIND_TYPE]
+            bind_as_user = bind_type == CONF_BIND_TYPE_AS_USER
+            username_attr = self.config[CONF_USERNAME_ATTR]
+
+            if bind_type == CONF_BIND_TYPE_ANONYMOUS:
+                _LOGGER.debug("Binding anonymously")
+                conn = ldap3.Connection(server, auto_bind=False)
+            else:
+                bind_dn = (
+                    f"{username_attr}={username},{base_dn}"
+                    if bind_as_user
+                    else self.config[CONF_BIND_DN]
+                )
+                bind_password = (
+                    password if bind_as_user else self.config[CONF_BIND_PASSWORD]
+                )
+                _LOGGER.debug("Binding as %s", bind_dn)
+                # LDAP bind
+                conn = ldap3.Connection(
+                    server, user=bind_dn, password=bind_password, auto_bind=False,
+                )
+            conn.open(read_server_info=False)
+            # Upgrade connection with START_TLS if requested.
+            if encryption == CONF_ENCRYPTION_STARTTLS:
+                conn.starttls(read_server_info=False)
```

Comment:
Replace with `conn.start_tls` (`conn.starttls` does not exist)

Reply:
Fixed in 2472dfb570a31c43d09a20cf227f8901f1cdf0e6